common/shield_exceptions.cc

```diff
@@ -40,69 +43,59 @@ bool IsBlockedResource(const GURL& gurl) {
 
 bool IsWhitelistedReferrer(const GURL& firstPartyOrigin,
     const GURL& subresourceUrl) {
-  // Note that there's already an exception for TLD+1, so don't add those here.
-  // Check with the security team before adding exceptions.
-
-  // https://github.com/brave/browser-laptop/issues/5861
-  // The below patterns are done to only allow the specific request
-  // pattern, of reddit -> redditmedia -> embedly -> imgur.
-  static auto redditPtrn = URLPattern(URLPattern::SCHEME_HTTPS,
-      "https://www.reddit.com/*");
-  static std::vector<URLPattern> reddit_embed_patterns({
-    redditPtrn,
-    URLPattern(URLPattern::SCHEME_HTTPS, "https://www.redditmedia.com/*"),
-    URLPattern(URLPattern::SCHEME_HTTPS, "https://cdn.embedly.com/*"),
-    URLPattern(URLPattern::SCHEME_HTTPS, "https://imgur.com/*")
-  });
-
-  if (redditPtrn.MatchesURL(firstPartyOrigin)) {
-    bool is_reddit_embed = std::any_of(
-      reddit_embed_patterns.begin(),
-      reddit_embed_patterns.end(),
-      [&subresourceUrl](URLPattern pattern){
-        return pattern.MatchesURL(subresourceUrl);
-      });
-    if (is_reddit_embed) {
-      return true;
-    }
-  }
-
-  static std::map<GURL, std::vector<URLPattern> > whitelist_patterns_map = {
-    {
-      GURL("https://www.facebook.com/"), {
-        URLPattern(URLPattern::SCHEME_HTTPS, "https://*.fbcdn.net/*"),
+  std::unique_ptr<base::Value> root = base::JSONReader::Read(
+    "{"
+    "    \"whitelist\": ["
```

Comment:
Might be nicer to use a raw string literal here, given all of the backslashes.

Reply:
This was an interim draft for testing the JSON parsing. The final version reads it from an external file.